### DISCUSSION TITLE
I get yt-dlp error

### DISCUSSION THREAD

**Discussion by suzumiya39** [atauenis/webone#121] [Q&A]
I get new error yt-dlp
Console says "Error ffmpeg with code 251"

Edit: Accoding to debug, It is faild to resolve host.
But Look my system network is fine. So, I tried minimam options.
I removed "-o -" option, "pipe:" options and change "best"  to "bestvideo+bestaudio".
Also, It is command succes.
How to fix this? I can't write C#.

**Comment by atauenis** [comment]
Tested now, it's working, but with some minor warning messages in log:
```
[youtube] Extracting URL: https://www.youtube.com/watch?v=fPnO26CwqYU
[youtube] fPnO26CwqYU: Downloading webpage
[youtube] fPnO26CwqYU: Downloading ios player API JSON
[youtube] fPnO26CwqYU: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Precondition check failed.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (1/3)...
[youtube] fPnO26CwqYU: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Precondition check failed.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (2/3)...
[youtube] fPnO26CwqYU: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Precondition check failed.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)...
[youtube] fPnO26CwqYU: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Precondition check failed.
WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: Bad Request>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
02.04.2024 11:08:46.211+25418030        <Send video/webm stream.
[youtube] fPnO26CwqYU: Downloading m3u8 information
[info] fPnO26CwqYU: Downloading 1 format(s): 22
[download] Destination: -

[download]   0.0% of    9.18MiB at  Unknown B/s ETA Unknown
[download]   0.0% of    9.18MiB at    2.93MiB/s ETA 00:03
[download]   0.1% of    9.18MiB at    3.41MiB/s ETA 00:02
[download]   0.2% of    9.18MiB at    7.32MiB/s ETA 00:01
[download]   0.3% of    9.18MiB at    1.89MiB/s ETA 00:04
[download]   0.7% of    9.18MiB at    1.82MiB/s ETA 00:04
[download]   1.4% of    9.18MiB at    1.86MiB/s ETA 00:04
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'pipe:':
```

Seems that YouTube is again introducing anti-downloading protection. But yt-dlp is still able to bypass it. At least, sometimes.

> I removed "-o -" option, "pipe:" options

It's required to work with WebOne & FFmpeg. By default, Yt-dlp is downloading into a one file or two files, then merges to one file. It's not suitable for on-fly converting or access via Web browser. So default options are configuring Yt-dlp to export single downloaded video+audio stream to FFmpeg input stream, then FFmpeg is outputting it to WebOne, which is redirecting the content to client.

> and change "best" to "bestvideo+bestaudio".

Strange, but this working with CLI version of yt-dlp, but not with WebOne. Probably yt-dlp is not correctly merging both DASH streams to a single with these parameters being used.

**Reply by suzumiya39** [reply]
thank you.
I try late, and I will show you log file.
